be/src/olap/field.h

```diff
@@ -55,9 +55,9 @@ class Field {
     inline size_t field_size() const { return size() + 1; }
     inline size_t index_size() const { return _index_size; }
 
-    inline void set_to_max(char* buf) const { return _type_info->set_to_max(buf); }
+    virtual inline void set_to_max(char* buf) const { return _type_info->set_to_max(buf); }
     inline void set_to_min(char* buf) const { return _type_info->set_to_min(buf); }
-    inline char* allocate_value_from_arena(Arena* arena) const { return _type_info->allocate_value_from_arena(arena); }
+    virtual inline char* allocate_value_from_arena(Arena* arena) const { return _type_info->allocate_value_from_arena(arena); }
```

Comment:
I think we can remove allocate_value_from_arena from TypeInfo, instead we can allocate memory here.

Reply:
But Field has no info about sizeof(CppType),only TypeInfo knows it

Reply:
we can get it through _type_info->size()

Reply:
👌